README updates

Andriciuc opened the pull request:
Updated the pg_tde README based on @Robinyo feedback, removed unnecessary config steps and replaced with links to docs for up to date information, reorganized content and information for clarity and descriptions, improved SEO.

Andriciuc committed:
Update README.md

updated readme to bring it in line with the latest RC2 updates

dAdAbird commented on contrib/pg_tde/README.md:
```diff
@@ -1,67 +1,77 @@
 # pg_tde: Transparent Database Encryption for PostgreSQL
 
-The PostgreSQL extension provides data at rest encryption. It is currently in an experimental phase and is under active development. [We need your feedback!](https://github.com/percona/pg_tde/discussions/151)
+The PostgreSQL extension provides data at rest encryption. It is currently in an experimental phase and is under active development. [We need your feedback!](https://github.com/percona/postgres/tree/TDE_REL_17_STABLE)
```
Shouldn't be here a link to discussions?

Andriciuc replied:
Thank you for that! The previous link was broken, now I updated it to the correct percona/postgres discussion.

dAdAbird commented on contrib/pg_tde/README.md:
```diff
@@ -1,67 +1,77 @@
 ## Overview
+
 Transparent Data Encryption offers encryption at the file level and solves the problem of protecting data at rest. The encryption is transparent for users allowing them to access and manipulate the data and not to worry about the encryption process. As a key provider, the extension supports the keyringfile and  [Hashicorp Vault](https://www.vaultproject.io/).
 
```
and KMIP

Andriciuc replied:
Fixed! Now reads:

As a key provider, the extension supports [keyringfile and external Key Management Systems (KMS) through a Global Key Provider interface](../pg_tde/documentation/docs/global-key-provider-configuration/index.md)

dAdAbird commented on contrib/pg_tde/README.md:
```diff
@@ -1,67 +1,77 @@
 
 ## Building from sources for community PostgreSQL
-  1. Install required dependencies (replace XX with 16 or 17)
+
```
pg_tde no longer works with community postgres it has to be our distribution.

Andriciuc replied:
I removed this whole thing and redirected users to our docs

dAdAbird reviewed (commented):
Does it make sense to have the installation section here at all? Let's maybe have a link to our docs for that? Otherwise, we would have to maintain the same info in two places, which is prone to errors.

Andriciuc committed:
Update README.md

removed building for community PG, updated discussions link, updated overview with links to our docs

Andriciuc commented:
> Does it make sense to have the installation section here at all? Let's maybe have a link to our docs for that? Otherwise, we would have to maintain the same info in two places, which is prone to errors.

True, it does not, removed this section as the readme already points to our docs

Andriciuc committed:
Update README.md

left the extension as tde is not a key provider

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into docs-readme-update
